### PR TITLE
Add http and https agent support

### PR DESCRIPTION
**Pull request by bchengwfd**
Add support for http and https-agent in Axios

### CODE REVIEW

File: src/bynder-js-sdk.js [hunk 1]
```diff
@@ -38,15 +38,19 @@ class APICall {
...
+     * @param {string} httpsAgent - A https agent.
+     * @param {string} httpAgent - A http agent.
      * @param {string} url - A string with the name of the API method.
      * @param {string} method - A string with the method of the API call.
      * @param {Object} consumerToken - An object with both the public and secret consumer keys.
      * @param {Object} accessToken - An object with both the public and secret access keys.
      * @param {Object} [data={}] - An object containing the query parameters.
      */
-    constructor(baseURL, url, method, consumerToken, accessToken, data = {}) {
+    constructor(baseURL, httpsAgent, httpAgent, url, method, consumerToken, accessToken, data = {}) {
```
Comment (GuilhermeSCP) [inline src/bynder-js-sdk.js]: These changes, the way they are implemented, will bring incompatibilities with previous versions of the SDK.
Please move the new parameters to the end of the function declaration.

Comment (GuilhermeSCP) [reply src/bynder-js-sdk.js]: Well, actually nope, somehow I ignored the rest of the file where you add the agent to *all* the calls 😄
You could indeed have added it to the end of the declaration but in theory it won't cause any problem.

**Comment by osener** [comment]
@bchengwfd thanks for your contribution! I think this PR highlights the huge amount of repetition in `APICall` class. What do you think about refactoring these two classes so that it is possible to create an instance of `APICall` in the constructor of `Bynder`? Something along the lines of:

```js
class Bynder {
  constructor(options) {
    this.api = new API({
      baseUrl,
      httpAgent,
      httpsAgent,
      consumerToken,
      accessToken
    });
  }

  getCollection({ id }) {
    this.api.get(`v4/collections/${id}`);
  }

  addMediaToCollection({ id, data }) {
    this.api.post(`v4/collections/${id}/media/`, { data });
  }
  ...
}
```

**Comment by bchengwfd** [comment]
Hi @osener

I have done some refactoring to the Bynder and APICall class without changing the code structure too much to make it a bit more tidy.

**Comment by osener** [comment]
This is great, thank you!

**Pull request merged**